test/e2e/federate.go

```diff
@@ -168,11 +172,75 @@ var _ = Describe("Federate ", func() {
 		}
 
 		By("Comparing the test resources with the templates of corresponding federated resources for equality")
-		validateResourcesEquality(tl, testResources, kubeConfig)
+		validateResourcesEqualityFromAPI(tl, createdTargetResources, kubeConfig)
+	})
+
+	It("input yaml from a file, should emit equivalant federated resources", func() {
+		tmpFile, err := os.Create(fmt.Sprintf("/tmp/file-%s", uuid.New()))
```

Comment:
Please use [ioutil.TempFile](https://golang.org/pkg/io/ioutil/#TempFile) to create a temporary file.  It is not safe to assume that `/tmp` is a safe location to write temporary files.  In my development environment, `/tmp` is not writeable by me and I use the `TMPDIR` env var to point a location I can write to.

Reply:
Updated!